integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CacheUtil.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command.cache
+
+import org.apache.hadoop.mapred.JobConf
+import scala.collection.JavaConverters._
+
+import org.apache.carbondata.core.cache.CacheType
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.core.datamap.Segment
+import org.apache.carbondata.core.metadata.schema.table.CarbonTable
+import org.apache.carbondata.core.readcommitter.LatestFilesReadCommittedScope
+import org.apache.carbondata.processing.merger.CarbonDataMergerUtil
+
+object CacheUtil {
+
+  def getAllIndexFiles(carbonTable: CarbonTable): List[String] = {
+    if (carbonTable.isTransactionalTable) {
+      val absoluteTableIdentifier = carbonTable.getAbsoluteTableIdentifier
+      CarbonDataMergerUtil.getValidSegmentList(absoluteTableIdentifier).asScala
+        .map {
+          segment =>
+            segment.getCommittedIndexFile
+        }
+        .flatMap {
+          indexFilesMap =>
+            indexFilesMap.keySet().toArray
+        }
+        .map(_.toString)
+        .toList
+    } else {
+      val tablePath = carbonTable.getTablePath
+      val readCommittedScope = new LatestFilesReadCommittedScope(tablePath, new JobConf())
+      readCommittedScope.getSegmentList
```

Comment:
same as the above logic can be changed to 

`readCommittedScope.getSegmentList
        .flatMap {
          load =>
            val seg = new Segment(load.getLoadName, null, readCommittedScope)
            seg.getCommittedIndexFile.keySet().asScala
        }.toList`

Reply:
Done.